### ISSUE TITLE
Crash when I use function tool

### ISSUE THREAD

**Issue by Evanineo** [zowepsilon/Functional#2]
My game just crashes when I use the function tool. Mods I'm using: Better Color Picker, BetterInfo, BetterInputs, Clean Startpos, Colorful Icons, Custom Keybinds, Editor Level ID API, EditorTab API, EvenMoreLevelTags, Functional (of course), Globed, Misc Bugfixes, Node IDs, Object Workshop, Redo Checkpoint, SkipEditorObjects, That *** Paste Button, Unlisted Objects in Editor, and some 2.206 mods

**Comment by zowepsilon** [comment]
please provide a crash log

**Comment by Evanineo** [comment]
2024-12-07T12:07:28-0500
Whoopsies! An unhandled exception has occurred.
It appears that the crash occurred while executing code from the "anatom3000.functional" mod. Please submit this crash report to its developers (anatom3000) for assistance.

== Geode Information ==
Loader Version: v4.0.1
Loader Commit: ed97f3b
Bindings Commit: e30b176
Installed mods: 23
Outdated mods: 4
Problems: 0

== Exception Information ==
Faulty Module: C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\geode\unzipped\anatom3000.functional\anatom3000.functional.dll
Faulty Mod: Functional v1.1.3 (anatom3000.functional)
Exception Code: c0000005 (EXCEPTION_ACCESS_VIOLATION)
Exception Flags: 0
Exception Address: 00007FFC8FDFE434 (anatom3000.functional.dll + 1e434)
Number Parameters: 2
Crashed thread: Main

== Stack Trace ==
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\geode\unzipped\anatom3000.functional\anatom3000.functional.dll + 1e434
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + 597cf (cocos2d::CCMenuItem::activate + 47)
 - 00000264728A0882 (Hook handler)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\Geode.dll + 3cf95 (geode::modifier::AsStaticFunction_activate<hook7<(anonymous namespace)::hook7Parent>,void (hook7<(anonymous namespace)::hook7Parent>::*)()>::Impl<void (hook7<(anonymous namespace)::hook7Parent>::*)()>::function + 21 | /home/runner/work/geode/geode/build/bindings/bindings/Geode/modify/CCMenuItem.hpp:35)
 - 00000264728A0882 (Hook handler)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + 58798 (cocos2d::CCMenu::ccTouchEnded + 40)
 - 00000264728A0C82 (Hook handler)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\Geode.dll + 3d65f (geode::modifier::AsStaticFunction_ccTouchEnded<hook15<(anonymous namespace)::hook15Parent>,void (hook15<(anonymous namespace)::hook15Parent>::*)(cocos2d::CCTouch *, cocos2d::CCEvent *)>::Impl<void (hook15<(anonymous namespace)::hook15Parent>::*)(cocos2d::CCTouch *, cocos2d::CCEvent *)>::function + 47 | /home/runner/work/geode/geode/build/bindings/bindings/Geode/modify/CCMenu.hpp:105)
 - 00000264728A0C82 (Hook handler)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + b0f2c (cocos2d::CCTouchDispatcher::touches + 444)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + 6b131 (cocos2d::CCEGLViewProtocol::handleTouchesEnd + 97)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + 76249 (cocos2d::CCEGLView::onGLFWMouseCallBack + 345)
 - 0000026406073C8B (Hook handler)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\geode\unzipped\spaghettdev.betterinputs\spaghettdev.betterinputs.dll + fdd4
 - 0000026406073C8B (Hook handler)
 - 0000026406073C8B (Hook handler)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + 76286
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + d6c1b
 - C:\WINDOWS\System32\USER32.dll + 183f1
 - C:\WINDOWS\System32\USER32.dll + 17c3b (CallWindowProcW + 139)
 - C:\WINDOWS\SYSTEM32\OPENGL32.dll + 41eb4
 - C:\WINDOWS\System32\USER32.dll + 183f1
 - C:\WINDOWS\System32\USER32.dll + 17eb1
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + d4548
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\libcocos2d.dll + 72c16 (cocos2d::CCApplication::run + 1078)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\GeometryDash.exe + 4b6e32
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\Geode.dll + 2402ad (gdMainHook + 557 | /home/runner/work/geode/geode/loader/src/platform/windows/main.cpp:181)
 - C:\Program Files (x86)\Steam\steamapps\common\Geometry Dash\GeometryDash.exe + 4b821a
 - C:\WINDOWS\System32\KERNEL32.DLL + 1257d (BaseThreadInitThunk + 29)
 - C:\WINDOWS\SYSTEM32\ntdll.dll + 5af08 (RtlUserThreadStart + 40)

== Register States ==
RAX: 0000000000000000
RBX: 0000000000000000
RCX: 0000026431e2be60
RDX: 000002643e7b8010
RBP: 0000000000000000
RSP: 000000f4ec1df980
RDI: 0000000000000000
RSI: 0000000000000000
RIP: 00007ffcc520af08
R8:  000002643dbd8c20
R9:  0000000000000001
R10: 0000000000000004
R11: 000002643e7b8010
R12: 0000000000000000
R13: 0000000000000000
R14: 0000000000000000
R15: 0000000000000000

== Installed Mods ==
x | [v1.0.7] alphalaneous.editortab_api
x | [v1.0.2] alphalaneous.that_paste_button
x | [v1.1.1] anatom3000.bettercolorpicker
x | [v1.1.3] anatom3000.functional
x | [v1.2.6] blueblock6.cleanstartpos
* | [v1.1.0] bobby_shmurner.zoom
* | [v1.0.3] cgytrus.level-shaders-fix
x | [v4.3.0-beta.4] cvolton.betterinfo
x | [v1.1.0] cvolton.level-id-api
x | [v1.3.1] cvolton.misc_bugfixes
x | [v1.7.2] dankmeme.globed2
* | [v2.0.0-alpha.3] elnexreal.auto_high_priority
x | [v1.4.4] firee.object-workshop
x | [v1.10.1] geode.custom-keybinds
x | [v4.0.1] geode.loader
x | [v1.18.0] geode.node-ids
x | [v1.3.5] legowiifun.unlisted_objects_in_editor
x | [v1.0.5] ninkaz.colorful-icons
x | [v1.1.2] raydeeux.skipeditorobjects
x | [v1.3.3] raydeeux_thesillydoggo.evenmoreleveltags
x | [v4.2.0-beta.12] spaghettdev.betterinputs
x | [v1.0.2] terma.redocheckpoint
* | [v1.12.7] undefined0.gdtweaks

sorry

**Comment by zowepsilon** [comment]
does it crash consistently or did it happen once?

**Comment by Evanineo** [comment]
It kept crashing when I used the function tool on spikes, but when I tried using it on that other thingy in the picture it didn't crash.
![confusing](https://github.com/user-attachments/assets/b5360551-1b48-48e7-965f-f3588669208c)
It seems like for certain objects it always crashes, and for certain objects it never crashes.

This isn't related to the crashes, but I also want to point out that the text shadow in the function tool isn't the same as in the other editor buttons.
